### PR TITLE
Arg summary

### PR DESCRIPTION
**Pull request by Luu-uuu**
Additional arguments will parse match summary only using Clarity package.

Needed to change dependency version on Snappy due to Apple Silicon.

### CODE REVIEW

File: src/main/java/opendota/Parse.java [hunk 1]
```diff
@@ -201,6 +202,24 @@ public Parse(InputStream input, OutputStream output) throws IOException
...
+      greevilsGreedVisitor = new GreevilsGreedVisitor(name_to_slot);
+      trackVisitor = new TrackVisitor();
+
+      isPlayerStartingItemsWritten = new ArrayList<>(Arrays.asList(new Boolean[numPlayers]));
+      Collections.fill(isPlayerStartingItemsWritten, Boolean.FALSE);
+      os = output;
+      long tStart = System.currentTimeMillis();
+
+      CDemoFileInfo info = Clarity.infoForFile(input);
+      System.out.println(info);
```
Comment (chudooder) [inline src/main/java/opendota/Parse.java]: Is this json formatted? we might want to make sure it is before node consumes it

File: src/main/java/opendota/Parse.java [hunk 2]
```diff
@@ -201,6 +202,24 @@ public Parse(InputStream input, OutputStream output) throws IOException
       long tMatch = System.currentTimeMillis() - tStart;
       System.err.format("total time taken: %s\n", (tMatch) / 1000.0);
     }
+
+    public Parse(String input, OutputStream output, boolean quick) throws IOException, InterruptedException
```
Comment (chudooder) [inline src/main/java/opendota/Parse.java]: Make the other constructor call this one with `quick = false`; then branch in this constructor. Also, you don't have to initialize a lot of the stuff if you're doing a quick parse

```java
public Parse(InputStream input, OutputStream output) throws IOException
{
  return Parse(input, output, false); 
}

public Parse(String input, OutputStream output, boolean quick) throws IOException, InterruptedException
{
  long tStart = System.currentTimeMillis();
  if (quick) {
    CDemoFileInfo info = Clarity.infoForFile(input);
    System.out.println(info);
  } else {
    greevilsGreedVisitor = new GreevilsGreedVisitor(name_to_slot);
    trackVisitor = new TrackVisitor();
    
    is = input;
    os = output;
    isPlayerStartingItemsWritten = new ArrayList<>(Arrays.asList(new Boolean[numPlayers]));
    Collections.fill(isPlayerStartingItemsWritten, Boolean.FALSE);
    long tStart = System.currentTimeMillis();
    new SimpleRunner(new InputStreamSource(is)).runWith(this);
  }
  long tMatch = System.currentTimeMillis() - tStart;
  System.err.format("total time taken: %s\n", (tMatch) / 1000.0);
}
```

File: src/main/java/opendota/Main.java [hunk 3]
```diff
@@ -18,9 +18,14 @@ public static void main(String[] args) throws Exception {
             FileInputStream is = new FileInputStream(file);
             OutputStream os = System.out;
             try {
-                new Parse(is, os);
-            } catch (Exception e) {
-                e.printStackTrace();
+		if (args.length >=2){
+			is.close();
+			new Parse(filepath,os,true);
+		}else{
```
Comment (chudooder) [inline src/main/java/opendota/Main.java]: nit: formatting `} else {`

File: src/main/java/opendota/Parse.java (hunk 1 above)
Comment (Luu-uuu) [reply src/main/java/opendota/Parse.java]: It is JSON formatted. New PR commit will resolve the extraneous info dumped by info. :)

**Review by chudooder — APPROVED** [review]

**Pull request merged**